application-flashmessages-ui/src/main/resources/Flash/FlashNotifier.xml

```diff
@@ -153,51 +153,70 @@
       <cache>long</cache>
     </property>
     <property>
-      <code>/**
- * Confirm that the user has seen the popups
- */
-window.flashNotifierConfirm = function(params){
-  window.flashPopup.closeDialog();
-  new Ajax.Request("$doc.getURL('get')", {
-    parameters : {'ajax' : '1', 'flash': params['document']},
-    method: 'get'
+      <code>require(['jquery'], function($) {
+  /**
+   * Initialize the flash notifier.
+   */
+  window.flashNotifier = function(params) {
+    window.flashNotifierParams = params;
+    window.flashNotifierIndex = 0;
+    window.flashNotifierNext();
+  };
+  /**
+  * Display the next popup.
+  */
+  window.flashNotifierNext = function() {
+    // If index &gt; length, stop it!
+    if(window.flashNotifierIndex &gt;= window.flashNotifierParams.length) {
+      return;
+    }
+    // Get the params.
+    var params = window.flashNotifierParams[window.flashNotifierIndex];
+    // Show the popup.
+    $(window).on('load', function() {
+      window.flashPopup.modal('show');
+    });
+    // When window has already been loaded.
+    if(window.flashNotifierIndex &gt; 0) {
+      window.flashPopup.modal('show');
+    }
+  }
+  /**
+   * Confirm that the user has seen the popups.
+   */
+  window.flashNotifierConfirm = function() {
+    let params = window.flashNotifierParams[window.flashNotifierIndex];
+    $.ajax({
+      url : "$doc.getURL('get')",
```

Comment:
You could use ``XWiki.currentDocument.getURL('get')`` to avoid the need to use Velocity.

Reply:
Because it's not always about the current document it's ok to use `new XWiki.Document('FlashNotifier', 'Flash').getURL('get')` ?

Reply:
Yes, for sure.